contracts/SimpleCollections.sol

```diff
@@ -117,14 +117,30 @@ contract SimpleCollections is ITokenURIDescriptor, OwnableUpgradeable {
 			return false;
 		}
 
-		// Always only allow staking via the SwapAndStake contract.
-		ISwapAndStake.Amounts memory stakeVia = swapAndStake.gatewayOf(
-			img.gateway
-		);
-
 		// Validate the staking position.
-		bool valid = img.requiredETHAmount <= stakeVia.input &&
-			img.requiredETHFee <= stakeVia.fee;
+		bool valid = false;
+
+		// `requiredETHFee` will be 0 when using direct `DEV` otherwise it's assumed
+		// that input currency is `ETH`.
+		if (img.requiredETHFee > 0) {
+			// This condition validates input user ETH and fee.
+			// Always only allow staking via the SwapAndStake contract.
+			ISwapAndStake.Amounts memory stakeVia = swapAndStake.gatewayOf(
+				img.gateway
+			);
+
+			valid =
+				img.requiredETHAmount <= stakeVia.input &&
+				img.requiredETHFee <= stakeVia.fee;
+		} else {
+			// This condition validates input `DEV` equivalent `ETH` with required.
+			// Fetch the `amount of ETH` the `DEV staked` is equal to.
+			uint256 equivalentETHStaked = swapAndStake.getEstimatedEthForDev(
```

Comment:
I am not sure if we can use this function. Since this uses
[`quoteExactInputSingle`](https://github.com/dev-protocol/eth-swap-and-stake/blob/main/contracts/SwapAndStakeV3.sol#L103) function which is a lens contract by uniswap meant only for off-chain calling since they are gas inefficient. [here](https://github.com/Uniswap/v3-periphery/blob/main/contracts/lens/Quoter.sol#L18)

Reply:
Nice catch. But how else can DEV:ETH equivalence be verified on-chain?